src/views/CreateView.vue

```diff
@@ -208,237 +209,221 @@
   </main>
 </template>
 
-<script>
-import { mapState } from 'pinia'
+<script setup lang="ts">
+import { ref, Ref, computed, onMounted } from 'vue'
+import { useRouter } from 'vue-router'
+import { useUserStore } from '../stores/user'
+import { handleModal } from '../composables/handleModal'
+import { supabase } from '../supabase'
+import { Exercise, ExerciseResponse, UserResponse } from '../api/types'
 import TheForm from '../components/shared/TheForm.vue'
 import TheInput from '../components/shared/TheInput.vue'
 import TheButton from '../components/shared/TheButton.vue'
 import TheExercise from '../components/shared/TheExercise.vue'
 import NeedSignIn from '../components/shared/NeedSignIn.vue'
 import TheToast from '../components/shared/TheToast.vue'
-import { useUserStore } from '../stores/user'
-import { supabase } from '../supabase'
 
-export default {
-  name: 'CreateView',
-  components: {
-    TheForm,
-    TheInput,
-    TheButton,
-    TheExercise,
-    NeedSignIn,
-    TheToast
-  },
-  data() {
-    return {
-      exercises: [],
-      workoutName: '',
-      exerciseName: '',
-      isNameShown: false,
-      areExercisesDisplayed: false,
-      selectedExercise: '',
-      kgValue: false,
-      lbValue: false,
-      exerciseData: {
-        name: '',
-        img: '',
-        muscles: '',
-        sets: '',
-        reps: '',
-        weight: '',
-        unit: '',
-        rest: ''
-      },
-      addedExercises: [],
-      isExerciseSubmitted: false,
-      isToastShown: false,
-      isInsertionError: false
+const exercises = ref<Exercise[]>([])
+const workoutName = ref('')
+const exerciseName = ref('')
+let selectedExercise: Ref<Exercise> | Ref<null> = ref(null)
+const exerciseData = ref<Exercise>({
+  name: '',
+  img: '',
+  muscles: [],
+  sets: '',
+  reps: '',
+  weight: '',
+  unit: '',
+  rest: ''
+})
+const addedExercises = ref<Exercise[]>([])
+const isExerciseSubmitted = ref(false)
+const isToastShown = ref(false)
+const isInsertionError = ref(false)
+
+const kgValue = ref(false)
+const lbValue = ref(false)
+const selectedUnit = computed(() => {
+  if (kgValue.value === true || lbValue.value === true) {
+    return kgValue.value === true ? 'kg' : 'lb'
+  }
+  return ''
+})
+
+const exercisesToShow = computed(() => {
+  const selectedExercises = showSelectedExercises(exerciseName.value)
+  
+  if (exerciseName.value === '') {
+    if (selectedExercise.value !== null) {
+      return exercises.value.filter((exercise) => selectedExercise.value === exercise)
     }
-  },
-  computed: {
-    ...mapState(useUserStore, ['isSignedIn']),
-    selectedUnit() {
-      if (this.kgValue === true || this.lbValue === true) {
-        return this.kgValue === true ? 'kg' : 'lb'
-      }
-      return ''
+    
+    return exercises.value
+  }
+  
+  if (selectedExercise.value !== null) {
```

Comment:
Same https://github.com/Kepmon/Workout-planner/pull/11/files#r1151510745

Reply:
**I guess it's corrected**